src/ncp/ncp_base_mtd.cpp

```diff
@@ -1431,6 +1428,12 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_STREAM_NET>(void)
     OT_UNUSED_VARIABLE(metaPtr);
     OT_UNUSED_VARIABLE(metaLen);
 
+    SuccessOrExit(error = otIp6GetPriority(mInstance, framePtr, frameLen, &msgSettings.mPriority));
```

Comment:
Rather than exposing a new API specific to getting the priority of an arbitrary pointer buffer + length, how about exposing something like `Message *otIp6NewMessageFromBuffer(const uint8_t *aData, uint16_t aDataLength)` that will take care of the priority and message allocation internally?

Reply:
Changed.